### PR TITLE
feat(client): provide more options when searching for opinions

### PR DESCRIPTION
**Pull request by ljgray**
Option to specify user and revision when searching for opinions. Also some black blank line format updates.

Note: these tests won't work until radiocosmology/caput#246 is merged

### CODE REVIEW

File: chimedb/dataflag/client.py [hunk 1]
```diff
@@ -338,14 +330,27 @@ def create_opinion(
...
+def opinion_list(type_, user, revision):
     """List known revisions of TYPE."""
 
     query = orm.DataFlagOpinion.select()
 
     if type_:
         query = query.where(orm.DataFlagOpinion.type == type_)
 
+    if user:
+        query = query.where(db.mediawiki.MediaWikiUser.user_name == user)
```
Comment (ketiltrout) [inline chimedb/dataflag/client.py]: Wiki usernames
```suggestion
    if user:
        user = user[0].upper() + user[1:]
        query = query.where(db.mediawiki.MediaWikiUser.user_name == user)
```

Wiki usernames are weird.  They're stored with an initial capital letter, but the capitalisation of the first letter is ignored when doing username look-ups.

File: chimedb/dataflag/client.py [hunk 2]
```diff
@@ -338,14 +330,27 @@ def create_opinion(
...
     help="Type of flagging opinion to list. If not set, list all opinions.",
 )
-def opinion_list(type_):
+@click.option(
+    "--user",
+    "-u",
+    help="Wiki user to search. If no user is supplied, all users are searched.",
+    default=None,
+)
+@click.option("--revision", "-r", type=REVISION, default=None)
```
Comment (ketiltrout) [inline chimedb/dataflag/client.py]: Needs a help string.

**Review by ketiltrout — CHANGES_REQUESTED** [review]
Matching Wiki user names is ridiculous.

File: chimedb/dataflag/client.py (hunk 1 above)
Comment (ljgray) [reply chimedb/dataflag/client.py]: Oh of course

**Review by ketiltrout — APPROVED** [review]

**Comment by ljgray** [comment]
Seems to be a temporary issue with github not being able to get some apt dependencies. I'll try to run the tests again in a few minutes

**Pull request merged**